### PR TITLE
Mss 901/update subscriptions lambda

### PR DESCRIPTION
**Pull request by NathanielBennett**
A new lambda, triggered by the Google subscription queue, consumes incoming messages and persists subscriptions to to dynamo

### CODE REVIEW

File: typescript/src/models/subscription.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        this.startTimeStamp = startTimeStamp;
+        this.endTimeStamp = endTimeStamp;
+        this.cancellationTimetamp = cancellationTimeStamp;
+        this.autoRenewing = autoRenewing;
+        this.googlePayload = googlePayload;
+        this.ttl = ttl;
+    }
+
+    get [DynamoDbTable]() {
+        return App + "-" + Stage + "-subscriptions"
```
Comment (webb04) [inline typescript/src/models/subscription.ts]: can also do: \`${App}-${Stage}-subscriptions\`

File: typescript/src/utils/aws.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    function () { return new ECSCredentials(); },
+    function () { return new SharedIniFileCredentials({
+        profile: "mobile"
+    }); }
+]);
+
+const aws = new DynamoDB({
+    region: Region,
+    credentialProvider: credentialProvider
+});
```
Comment (webb04) [inline typescript/src/utils/aws.ts]: You can also do this in JavaScript
```
const aws = new DynamoDB({
    region: Region,
    credentialProvider
});
```

File: typescript/src/updatesubs/google.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
+process.on('uncaughtException', function (err) { console.log(err); })
```
Comment (webb04) [inline typescript/src/updatesubs/google.ts]: Do we want this for logging?

File: typescript/src/utils/google-play.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import aws = require("../utils/aws");
```
Comment (webb04) [inline typescript/src/utils/google-play.ts]: Could you just import s3? Ah you have the s3 type below, so maybe this is clearer.

File: typescript/src/utils/aws.ts (hunk 2 above)
Comment (NathanielBennett) [reply typescript/src/utils/aws.ts]: I don't think it looks as clean though

File: typescript/src/updatesubs/google.ts (hunk 3 above)
Comment (NathanielBennett) [reply typescript/src/updatesubs/google.ts]: No I need to get rid of it

**Pull request merged**